examples/vmod_vfp/src/lib.rs

```diff
@@ -1,13 +1,45 @@
-varnish::boilerplate!();
-
 use std::ffi::CStr;
 
-use varnish::ffi;
-use varnish::vcl::ctx::{Ctx, Event};
-use varnish::vcl::processor::{new_vfp, InitResult, PullResult, VFPCtx, VFP};
-use varnish::vcl::vpriv::VPriv;
+use varnish::vcl::ctx::Ctx;
+use varnish::vcl::processor::{InitResult, PullResult, VFPCtx, VFP};
+
+#[varnish::vmod]
+mod vfp {
+    use varnish::ffi;
+    use varnish::vcl::ctx::{Ctx, Event};
+    use varnish::vcl::processor::new_vfp;
 
-varnish::vtc!(test01);
+    use crate::Lower;
```

Comment:
could we have kept `Lower` at the same level as the rest of the code?

Reply:
The whole module must be tagged with `#[vmod]` attribute in order for the proc-macro to modify it.  The macro currently looks for several signals:
* a public function -- treated as a function or an event if it has `#[event]`
* an `impl` block -- treated as methods (only allows non-trait impls)

This means that it is usually cleaner to separate Varnish interface from the rest of the code, and this pattern demos how to do that. We could have more code inside the `vfp` mod, but it might create some surprises or simply won't compile in some cases, e.g. some public fn without the right signature - so better teach best practices this way